io-engine/src/core/reactor.rs

```diff
@@ -624,3 +635,101 @@ impl Future for &'static Reactor {
         }
     }
 }
+
+/// Maximum number of heartbeats a reactor is allowed to miss
+/// before it is classified as frozen.
+#[cfg(feature = "diagnostics")]
+const MAX_MISSED_HEARTBEATS: u64 = 3;
```

Comment:
What's the rationale behind using a max missed hearbeats rather than simply bumping the interval to 3s and waiting for that to elapse once for example?